php-checks/src/main/java/org/sonar/php/checks/UseOfUninitializedVariableCheck.java

```diff
@@ -0,0 +1,279 @@
+    // "$this" is defined only in method, but rule S2014 raises issues when it's used elsewhere
+    "$this"));
+
+  private static final Set<String> FUNCTION_ALLOWING_ARGUMENT_CHECK;
+  static {
+    FUNCTION_ALLOWING_ARGUMENT_CHECK = new HashSet<>(IgnoredReturnValueCheck.PURE_FUNCTIONS);
+    FUNCTION_ALLOWING_ARGUMENT_CHECK.remove("isset");
+  }
+
+  private static final Map<Kind, Predicate<Tree>> IS_READ_ACCESS_BY_PARENT_KIND;
+  static {
+    Map<Kind, Predicate<Tree>> map = new EnumMap<>(Kind.class);
+
+    PARENT_INITIALIZATION_KIND.forEach(kind -> map.put(kind, tree -> false));
+    ASSIGNMENT_KIND.forEach(kind -> map.put(kind, tree -> tree == ((AssignmentExpressionTree) tree.getParent()).value()));
+    map.put(Kind.FUNCTION_CALL, tree -> {
+      FunctionCallTree functionCall = (FunctionCallTree) tree.getParent();
+      return tree == functionCall.callee() || FUNCTION_ALLOWING_ARGUMENT_CHECK.contains(lowerCaseFunctionName(functionCall));
+    });
+    map.put(Kind.FOREACH_STATEMENT, tree -> tree == ((ForEachStatementTree) tree.getParent()).expression());
+    map.put(Kind.ARRAY_ACCESS, tree -> !isArrayAssignment(tree));
+    map.put(Kind.PARENTHESISED_EXPRESSION, tree -> isReadAccess(tree.getParent()));
+
+    IS_READ_ACCESS_BY_PARENT_KIND = map;
+  }
+
+  @Override
+  public void visitFunctionDeclaration(FunctionDeclarationTree tree) {
+    checkFunction(tree);
+    super.visitFunctionDeclaration(tree);
+  }
+
+  @Override
+  public void visitFunctionExpression(FunctionExpressionTree tree) {
+    checkFunction(tree);
+    super.visitFunctionExpression(tree);
+  }
+
+  @Override
+  public void visitMethodDeclaration(MethodDeclarationTree tree) {
+    checkFunction(tree);
+    super.visitMethodDeclaration(tree);
+  }
+
+  private void checkFunction(FunctionTree function) {
+    VariableVisitor visitor = new VariableVisitor(function);
+    function.accept(visitor);
+    visitor.uninitializedStream()
+      .forEach(variable -> context().newIssue(this, variable, "Review the data-flow - use of uninitialized value."));
+  }
+
+  private static class VariableVisitor extends PHPVisitorCheck {
+
+    final FunctionTree currentFunction;
+
+    boolean trustVariables = true;
+
+    Map<String, VariableIdentifierTree> firstVariableReadAccess = new HashMap<>();
+
+    Set<String> initializedVariables = new HashSet<>();
+
+    VariableVisitor(FunctionTree currentFunction) {
+      this.currentFunction = currentFunction;
+    }
+
+    Stream<VariableIdentifierTree> uninitializedStream() {
+      if (!trustVariables) {
+        return Stream.empty();
+      }
+      return firstVariableReadAccess.entrySet().stream()
+        .filter(entry -> !initializedVariables.contains(entry.getKey()))
+        .map(Map.Entry::getValue);
+    }
+
+    @Override
+    public void visitVariableIdentifier(VariableIdentifierTree tree) {
+      if (isMemberAccess(tree)) {
+        return;
+      }
+      String lowerCaseName = tree.text().toLowerCase(Locale.ROOT);
```

Comment:
I think we shouldn't transform the name of the variable. Variable names are case-sensitive.